File: rust/26.remove-duplicates-from-sorted-array.rs
/*
 * @lc app=leetcode id=26 lang=rust
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        use std::collections::HashSet;

        let mut set: HashSet<i32> = nums.iter().cloned().collect();
        *nums = set.iter().cloned().collect();
        nums.sort();
        nums.len() as i32
    }
}
// @lc code=end

File: javascript/26.remove-duplicates-from-sorted-array.js
/*
 * @lc app=leetcode id=26 lang=javascript
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @ts-check

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  const set = new Set(nums);
  let i = 0;
  set.forEach((v) => (nums[i++] = v));
  return set.size;
};
// @lc code=end
